src/test/java/selenium/tutorial/studies/HTMLUnitDriverTest.java

```diff
@@ -0,0 +1,129 @@
+import java.io.InputStream;
+import java.util.Properties;
+import java.util.concurrent.TimeUnit;
+
+import org.openqa.selenium.Alert;
+import org.openqa.selenium.By;
+import org.openqa.selenium.WebDriver;
+import org.openqa.selenium.WebElement;
+import org.openqa.selenium.chrome.ChromeDriver;
+import org.openqa.selenium.htmlunit.HtmlUnitDriver;
+import org.openqa.selenium.interactions.Actions;
+import org.testng.Assert;
+import org.testng.annotations.AfterTest;
+import org.testng.annotations.BeforeTest;
+import org.testng.annotations.Test;
+
+import io.github.bonigarcia.wdm.WebDriverManager;
+
+public class HTMLUnitDriverTest {
+
+	
+	WebDriver driver;
+	/**
+	 * Will be executed before all methods annotated with @Test. 
+	 * We can give any method name but must be annotated with @BeoreTest if we want these codes to be executed before test case scenarions
+	 * Usually all initializing parameters like driver initialization and common weburl if being used in all testcase methods in the same class are defined here
+	 * @throws IOException
+	 */
+	@BeforeTest(enabled=true)
+	public void setup() throws IOException {
+		InputStream input=HTMLUnitDriverTest.class.getClassLoader().getResourceAsStream("testData.properties");
+		Properties property=new Properties();
+		property.load(input);
+		
+		
+		if("chrome".equalsIgnoreCase(property.getProperty("activeBrowser"))) {
+						
+			WebDriverManager.chromedriver().version(property.getProperty("driverVersion")).version("74.0").setup();
+			driver=new ChromeDriver(); 
+			//***Browser specific **
+			driver.manage().window().maximize();
+			driver.manage().deleteAllCookies();
+			driver.manage().timeouts().pageLoadTimeout(15, TimeUnit.SECONDS);
+			driver.manage().timeouts().implicitlyWait(15, TimeUnit.SECONDS);
+			//driver=new HtmlUnitDriver();
+		
+		} else if("firefox".equalsIgnoreCase(property.getProperty("activeBrowser"))) {
+			//Add mozilla specific driver and initialize the driver
+			
+			//***Browser specific **
+			driver.manage().window().maximize();
+			driver.manage().deleteAllCookies();
+			driver.manage().timeouts().pageLoadTimeout(15, TimeUnit.SECONDS);
+			driver.manage().timeouts().implicitlyWait(15, TimeUnit.SECONDS);
+		}
+		
+		else if("HTMLUnitDriver".equalsIgnoreCase(property.getProperty("activeBrowser"))) {
+			//Add HTMLUnitDriver and initialize the driver
+			
+			driver=new HtmlUnitDriver();
+			((HtmlUnitDriver)driver).setJavascriptEnabled(true);
+			//No need of Window maximise and timeout as no window will be opened
+			
+		}
+		
+		driver.get(property.getProperty("usermgmtAppURL")); 
+	}
+	
+	/**
+	 * Actual Testcase script for a particular scenario is written here.
+	 * We can create multiple methods annotated with @Test if there are multiple scenarios too be executed.
+	 * @throws InterruptedException 
+	 */
+	@Test(enabled=true)
+	public void testSingleClickHandling() throws InterruptedException {
+		System.out.println("Inside testSingleClickHandling()");
```

Comment:
Once your script is working, remove System.out.print() . this is used for when u r debugging an issue. But remove when u r finally pushing the code in git hub. Make changes where ever applicable

Reply:
Once your script is working, remove System.out.print() . this is used for when u r debugging an issue. But remove when u r finally pushing the code in git hub. Make changes where ever applicable